### PR TITLE
Category loading & secrets

### PR DESCRIPTION
**Pull request by ncuillery**
- First bloc and category loading
- Non-recorded bit: secrets handling

### CODE REVIEW

File: lib/category_list/apis/category_api.dart [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const DOMAIN = 'https://flex-cms-fpnplvnjqq-uc.a.run.app';
+const PATH = '/api/categories';
+
+class CategoryApi {
+  final http = Dio()
+    ..options.headers = {
+      'Authorization': 'Bearer ${dotenv.get('STRAPI_TOKEN')}',
+    };
+
+  Future<List<Category>> fetchCategories({int? parentId}) async {
```
Comment (stephenrichter) [inline lib/category_list/apis/category_api.dart:14]: Will parentId always be an `int` or would this be better served as a `String` in most cases?

File: lib/category_list/apis/category_api.dart [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    final response = await http.get(
+        '$DOMAIN$PATH?filters[parent][id]${parentId == null ? '[\$null]=true' : '[\$eq]=$parentId'}&populate=*');
+
+    print(response.statusCode);
+    print(response.data['data']);
```
Comment (stephenrichter) [inline lib/category_list/apis/category_api.dart]: Let's remove print statements before merge

File: lib/secrets.dart
Comment (stephenrichter) [inline lib/secrets.dart:1]: Can remove this file now that we use env variables

File: lib/shop/shop_page.dart [hunk 3]
```diff
@@ -1,5 +1,8 @@
 import 'package:auto_route/auto_route.dart';
+import 'package:flex_storefront/category_list/cubits/category_cubit.dart';
+import 'package:flex_storefront/category_list/cubits/category_state.dart';
```
Comment (stephenrichter) [inline lib/shop/shop_page.dart:3]: Do we want to export these in a single file? i.e. `category_cubit.dart` or always keep separate?

**Review by stephenrichter — CHANGES_REQUESTED** [review]

File: lib/shop/shop_page.dart (hunk 3 above)
Comment (ncuillery) [reply lib/shop/shop_page.dart:3]: I would go with the separated imports, barrel files are making my dev workflow harder TBH.

File: lib/category_list/apis/category_api.dart (hunk 1 above)
Comment (ncuillery) [reply lib/category_list/apis/category_api.dart:14]: If we move out of Strapi it possibly won't, but let's deal with it when the time will come (if it comes).

**Review by stephenrichter — APPROVED** [review]
Changes look good, thanks! ✅

**Pull request merged**